warehouse/query-core/src/main/java/datawave/query/jexl/visitors/ExpandCompositeTerms.java

```diff
@@ -173,8 +174,8 @@ public Object visit(ASTOrNode node, Object data) {
     public Object visit(ASTAndNode node, Object data) {
         ExpandData parentData = (ExpandData) data;
         
-        // ignore marked nodes
-        if (QueryPropertyMarkerVisitor.instanceOfAny(node))
+        // ignore marked nodes except delays
+        if (QueryPropertyMarkerVisitor.instanceOfAny(node) && !QueryPropertyMarkerVisitor.instanceOf(node, ASTDelayedPredicate.class, null))
```

Comment:
I think you could improve performance here by passing a list of acceptable markers instead of doing one pass checking for any, and another checking for delayed predicate.

Reply:
good idea